Fix assigning error or event selectors to constant variables. (#16325)
ethereum/solidity: Changelog.md

Compiler Features:

Bugfixes:

> * TypeChecker: Fix assigning error or event selectors to constant variables.

Reviewer: ```suggestion
* TypeChecker: Allow assignment of error and event selectors to constant variables.
```

Reviewer 2: This is not incorrect, but it's a symptom rather than the cause. IMO this would be more accurate:

```
* TypeChecker: Fix error and event selectors not being considered compile-time constant.
```
